src/solvers/smt2/smt2_parser.h

```diff
@@ -80,6 +80,19 @@ class smt2_parsert:public smt2_tokenizert
 
   /// Apply typecast to unsignedbv to given expression
   exprt cast_bv_to_unsigned(const exprt &);
+
+public:
+  class smt2_format
+  {
+  public:
+    explicit smt2_format(const typet &_type) : type(_type)
+    {
+    }
+
+    const typet type;
+  };
+
+  friend std::ostream &operator<<(std::ostream &, const smt2_format &);
```

Comment:
Why is `friend`-ship indicated here? Wouldn't a declaration in the global namespace just do the job? Or indeed actually a member `std::ostream &operator<<(std::ostream &) const` in `smt2_format`?

Reply:
Ok, now in the global namespace.